### ISSUE TITLE
DomReady does not work when it reloads page.

### ISSUE THREAD

**Issue by diegoflorez** [mootools/mootools-core#2682]
En firefox 34.0.5 when it reloads the page, the code inside

``` javascript
window.addEvent('domready',function(){
   //code inside
})
```

is never called again, until you close the window/tab and reopen the page.
It's imposible to make a jsfiddle to show this behaviour.
I thinks this is a bug. Because you are trying to reload the page, so the DOM is for a new instance, ready.
##

**Comment by ricardograca** [comment]
I think this is an issue with RequireJS interacting in a strange way with Mootools and causing some kind of race condition. Check this out if you need to run scripts after the dom is ready and you're using RequireJS: https://github.com/requirejs/domReady

**Comment by SergioCrisostomo** [comment]
@diegoflorez thank you for fixing the example showing the problem!
@ricardograca nice input! (obrigado!)

Could we fix this by just adding `if (!ready) check();` [in this line](https://github.com/SergioCrisostomo/mootools-core/blob/master/Source/Utilities/DOMReady.js#L104)? (last line of `DOMReady.js`)

**Comment by diegoflorez** [comment]
Hey Guys, thanks for all your work.
It works Perfect! for the domready problem.
But now It throws this error: "TypeError: window is null".

**Comment by SergioCrisostomo** [comment]
@diegoflorez could you share a jsFiddle reproducing that error? what line does it point to? 

I don't see how `window` can be  `null`.

**Comment by diegoflorez** [comment]
@SergioCrisostomo I have just made the change that you have suggested in fix-2682 to the mootools-core-5.1 no compat. Remove the line 6033 and Add from 6034 to 6038.
You can see that behaviour here: http://diegoflorez.com/workspace/mootools.test/dom.ready.test.html

**Comment by SergioCrisostomo** [comment]
I see now, nice with a demo page. So RequireJS is not providing the `window` object so MooTools cannot be loaded. I am not so at home with RequireJS but will take look at this again.

Maybe someone else more used to RequireJS will have some feedback soonish...

**Comment by diegoflorez** [comment]
You're right here is an example without RequireJS http://diegoflorez.com/workspace/mootools.test/dom.ready.fix-2682.test.html

Problem solved, I'll remove the demo page in a few days.

Thank you @SergioCrisostomo

**Comment by diegoflorez** [comment]
[Opening Again]

I have run 2 extra tests to this fix:
I have replaced the Mootools domReady and I have used the requireJS domReady, 
1. when using mootools.core.1.5.1.js it works perfect, 
(as you can see here: http://diegoflorez.com/workspace/mootools.test/dom.ready.test.html)
2. but when using mootools.core.1.5.1.fix-2682.js throws the error message "TypeError: window is null"
(as you can see here: http://diegoflorez.com/workspace/mootools.test/dom.ready.fix-2682.test.html)

**Comment by SergioCrisostomo** [comment]
Using RequireJS's own `domready` seems the logical way, and best practise, since its RequireJS the one to load dependencies (MooTools). That said this issue can be closed IMO. I will close it in some days if no one else has input on it.

As a side note I still think the fix is usefull, but that we can discuss in the pull request itself.

Thank you @diegoflorez for nice input and demos to analyse this problem.

**Comment by deadmann** [comment]
i'm not sure if it help or not,

but also i receive error on webstorm intelisense for this part:

line: 5848

    // Make sure that domready fires before load
    Element.Events.load = {
        base: 'load',
        onAdd: function(fn){
            if (loaded && this == window) fn.call(this);
        },
        condition: function(){
            if (this == window){
                domready(); ------Error is on this line -> invalid call target------
                delete Element.Events.load;
            }
            return true;
        }
    };

my version is 1.4.5 (get from nuget)
I don't see this error on minified version

**Comment by cocprobase** [comment]
I have never seen a more wonderful post than this one.  [All Best Town Hall 6 Base Ever](http://cocpro.org/)
